### PR TITLE
Static Code Analysis with Php Inspections (EA Extended)

### PR DESCRIPTION
**Pull request by kalessil**
UTs: passed
Non-functional changes, micro-optimizations

### CODE REVIEW

File: lib/cache/sfMemcacheCache.class.php [hunk 1]
```diff
@@ -60,7 +60,7 @@ public function initialize($options = array())
         foreach ($this->getOption('servers') as $server)
         {
           $port = isset($server['port']) ? $server['port'] : 11211;
-          if (!$this->memcache->addServer($server['host'], $port, isset($server['persistent']) ? $server['persistent'] : true))
+          if (!$this->memcache->addserver($server['host'], $port, isset($server['persistent']) ? $server['persistent'] : true))
```
Comment (GromNaN) [inline lib/cache/sfMemcacheCache.class.php]: This is correct [`addServer`](http://php.net/manual/en/memcache.addserver.php)

Comment (kalessil) [inline lib/cache/sfMemcacheCache.class.php]: Probably PhpStoms has incorrect definition in stubs, will revert this.

**Comment by GromNaN** [comment]
Replacing `rand` by `mt_rand` seems to be a [good idea](http://php.net/manual/en/function.mt-rand.php).

>  The mt_rand() function [...] will produce random numbers four times faster than what the average libc rand() provides.

File: lib/config/sfPluginConfiguration.class.php [hunk 2]
```diff
@@ -211,18 +211,18 @@ public function filterTestFiles(sfEvent $event, $files)
...
    * Guesses the plugin root directory.
-   * 
+   *
    * @return string
    */
   protected function guessRootDir()
   {
     $r = new ReflectionClass(get_class($this));
-    return realpath(dirname($r->getFilename()).'/..');
+    return realpath(dirname($r->getFileName()).'/..');
```
Comment (GromNaN) [inline lib/config/sfPluginConfiguration.class.php]: :+1: http://php.net/manual/en/reflectionclass.getfilename.php

**Comment by kalessil** [comment]
mt_* is basically new generation replacement. Hopefully I'll find time to write a PHP CS FIxer to automate this (like for is_null, alias functions, __DIR__ usage).

File: lib/config/sfPluginConfiguration.class.php (hunk 2 above)
Comment (kalessil) [inline lib/config/sfPluginConfiguration.class.php]: Actually, this found automatically, I spent couple our just fixing.

File: lib/request/sfRequest.class.php [hunk 3]
```diff
@@ -301,12 +301,9 @@ public function setParameter($name, $value)
    */
   public function getContent()
   {
-    if (null === $this->content)
+    if (null === $this->content && '' === trim($this->content = file_get_contents('php://input')))
```
Comment (GromNaN) [inline lib/request/sfRequest.class.php]: :+1:

File: lib/widget/sfWidgetFormSchema.class.php [hunk 4]
```diff
@@ -822,7 +822,7 @@ public function moveField($field, $action, $pivot = null)
         array_unshift($this->positions, $field);
         break;
       case sfWidgetFormSchema::LAST:
-        array_push($this->positions, $field);
+        $this->positions []= $field;
```
Comment (GromNaN) [inline lib/widget/sfWidgetFormSchema.class.php]: I used to write it like this: `$this->positions[] = $field`. But this syntax definitely makes sense.

**Comment by GromNaN** [comment]
Thanks, that's awesome. All that micro-optimizations will be rewarding for all our apps.
I'm curious to know how you did that.

**Comment by kalessil** [comment]
Static Code Analysis is my tool: https://plugins.jetbrains.com/plugin/7622 (you'll need PhpStorm for it).

And fixing done partially by this this fixers:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1304
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1302
https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/d86a8aa39cca292d1b9884c1431b3d5db48e6063

Quite lot of this is automated (not everything of course).

**Comment by kalessil** [comment]
SCA Tool is advanced one, but most probably you'll want to adjust it for your needs.

Feel free to ask questions )

**Comment by GromNaN** [comment]
Thanks @kalessil. I'll try it on my projects.

File: lib/cache/sfMemcacheCache.class.php (hunk 1 above)
Comment (GromNaN) [inline lib/cache/sfMemcacheCache.class.php]: Fixed. https://github.com/LExpress/symfony1/commit/1cb8f4539aeff38a7fa8a47f349ed839a0dbcc9c

Comment (kalessil) [inline lib/cache/sfMemcacheCache.class.php]: Thank you!

**Pull request merged**